iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/LoadDetailXmlSerializer.java

```diff
@@ -6,61 +6,10 @@
-import com.powsybl.commons.xml.XmlWriterContext;
-import com.powsybl.iidm.network.Load;
-
-import javax.xml.stream.XMLStreamException;
-
 /**
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ *
+ * @deprecated Use {@link com.powsybl.iidm.xml.extensions.LoadDetailXmlSerializer} instead.
  */
-@AutoService(ExtensionXmlSerializer.class)
-public class LoadDetailXmlSerializer extends AbstractExtensionXmlSerializer<Load, LoadDetail> {
-
-    public LoadDetailXmlSerializer() {
-        super("detail", "network", LoadDetail.class,
-                false, "loadDetail.xsd",
-                "http://www.itesla_project.eu/schema/iidm/ext/load_detail/1_0",
-                "ld");
-    }
-
-    @Override
-    public void write(LoadDetail detail, XmlWriterContext context) throws XMLStreamException {
-        XmlUtil.writeFloat("fixedActivePower", detail.getFixedActivePower(), context.getExtensionsWriter());
-        XmlUtil.writeFloat("fixedReactivePower", detail.getFixedReactivePower(), context.getExtensionsWriter());
-        XmlUtil.writeFloat("variableActivePower", detail.getVariableActivePower(), context.getExtensionsWriter());
-        XmlUtil.writeFloat("variableReactivePower", detail.getVariableReactivePower(), context.getExtensionsWriter());
-    }
-
-    @Override
-    public LoadDetail read(Load load, XmlReaderContext context) {
-        float fixedActivePower = XmlUtil.readOptionalFloatAttribute(context.getReader(), "fixedActivePower");
-        if (Float.isNaN(fixedActivePower)) {
-            fixedActivePower = XmlUtil.readFloatAttribute(context.getReader(), "subLoad1ActivePower");
-        }
-        float fixedReactivePower = XmlUtil.readOptionalFloatAttribute(context.getReader(), "fixedReactivePower");
-        if (Float.isNaN(fixedReactivePower)) {
-            fixedReactivePower = XmlUtil.readFloatAttribute(context.getReader(), "subLoad1ReactivePower");
-        }
-        float variableActivePower = XmlUtil.readOptionalFloatAttribute(context.getReader(), "variableActivePower");
-        if (Float.isNaN(variableActivePower)) {
-            variableActivePower = XmlUtil.readFloatAttribute(context.getReader(), "subLoad2ActivePower");
-        }
-        float variableReactivePower = XmlUtil.readOptionalFloatAttribute(context.getReader(), "variableReactivePower");
-        if (Float.isNaN(variableReactivePower)) {
-            variableReactivePower = XmlUtil.readFloatAttribute(context.getReader(), "subLoad2ReactivePower");
-        }
-        load.newExtension(LoadDetailAdder.class)
-                .withFixedActivePower(fixedActivePower)
-                .withFixedReactivePower(fixedReactivePower)
-                .withVariableActivePower(variableActivePower)
-                .withVariableReactivePower(variableReactivePower)
-                .add();
-        return load.getExtension(LoadDetail.class);
-    }
+public class LoadDetailXmlSerializer extends com.powsybl.iidm.xml.extensions.LoadDetailXmlSerializer {
```

Comment:
I'm not sure it's useful (or it's working...):
- People that need this serializer needs powsybl-iidm-xml-converter, so they automatically have the serializer
- You will have 2 serializer for the same extension: is it working?

I propose to simply move the class without managing the compatibility at compile time. At runtime, it will work perfectly.

Reply:
You're right, it is only ever used in runtime so keeping it is useless.